seastar/src/clocks.rs

```diff
@@ -550,13 +551,139 @@ impl<ClockType> Sub for Instant<ClockType> {
     }
 }
 
+#[doc(hidden)]
+pub enum CppTimer {
+    Steady(UniquePtr<steady_clock_timer>),
+    Lowres(UniquePtr<lowres_clock_timer>),
+    Manual(UniquePtr<manual_clock_timer>),
+}
```

Comment:
Hmm, why do we have a single `CppTimer`? I thought we would have a separate Rust type for each seastar clock. You could then expose this type via associated type in `ClockImpl` so that the generic `Timer` that you introduce in the next commit will have access to it.

Reply:
I fixed it.